unit-tests/unit-test-config.py

```diff
@@ -172,7 +172,11 @@ def process_cpp( dir, builddir ):
         testdir = os.path.splitext( f )[0]                          # "log/internal/test-all"  <-  "log/internal/test-all.cpp"
         testparent = os.path.dirname(testdir)                       # "log/internal"
         # We need the project name unique: keep the path but make it nicer:
-        testname = 'test-' + testparent.replace( '/', '-' ) + '-' + os.path.basename(testdir)[5:]   # "test-log-internal-all"
+        if testparent:
+            testname = 'test-' + testparent.replace( '/', '-' ) + '-' + os.path.basename( testdir )[
+                                                                        5:]  # "test-log-internal-all"
+        else:
+            testname = testdir  # no parent folder so we get "test-all"
```

Comment:
This is similar to the edge case of a test being directly in the `unit-tests` folder in `run-unit-tests`